lib/std/posix.zig

```diff
@@ -4276,6 +4276,34 @@ pub fn connect(sock: socket_t, sock_addr: *const sockaddr, len: socklen_t) Conne
     }
 }
 
+pub const GetSockOptError = error{
+    /// The option is not supported by the protocol.
+    InvalidProtocolOption,
+
+    /// Insufficient resources are available in the system to complete the call.
+    SystemResources,
+
+    /// The file descriptor is not valid.
+    FileDescriptorNotValid,
+
+    // The file descriptor is not a socket.
+    FileDescriptorNotASocket,
```

Comment:
... and I'd like to say the same about this one, but I see that we currently treat it as a legitimate error throughout the module. It seems to me that that ought to change, but I think that's out of scope for this PR.

Reply:
Actually, no, the `setsockopt()` function does treat this as programmer error. So let's be consistent here and also make this an `unreachable` prong.